client.go

```diff
@@ -30,6 +31,7 @@ type Client struct {
 	EncodingBase64 bool
 	EventID        string
 	subscribed     map[chan *Event]chan bool
+	mu             sync.Mutex
```

Comment:
It would be better to embed the mutex here, so:

```
type Client struct {
	EncodingBase64 bool
	EventID        string
	subscribed     map[chan *Event]chan bool
	
        sync.Mutex
```